src/ck-core/init.C

```diff
@@ -914,20 +918,19 @@ static inline void _processRODataMsg(envelope *env)
 {
   //Unpack each readonly:
   if(!CmiMyRank()) {
-#if CMK_ONESIDED_IMPL && CMK_SMP
-    if(CMI_IS_ZC_BCAST(env)) {
-      // Send message to peers
-      CmiForwardMsgToPeers(env->getTotalsize(), (char *)env);
-    }
-#endif
-
     //Unpack each readonly
     PUP::fromMem pu((char *)EnvToUsr(env));
     CmiSpanningTreeInfo &t = *_topoTree;
 
 #if CMK_ONESIDED_IMPL
-    pu|numZerocopyROops;
+#if CMK_SMP
+    UInt numZerocopyROopsTemp;
```

Comment:
Just to clarify, this is necessary because pup doesn't support atomic?

Reply:
also, would it be possible/easier to do a cast, ie, `pu|(Uint) numZerocopyROOps` (or call `pup()` explicitly)?

Reply:
Yeah, I added because it seemed like pup didn't support atomics. 

```
init.C:932:9: error: invalid conversion from 'std::__atomic_base<unsigned int>::__int_type' {aka 'unsigned int'} to 'mCastEntryPtr' {aka 'mCastEntry*'} [-fpermissive]
     pu|((UInt)numZerocopyROops);
        ~^~~~~~~~~~~~~~~~~~~~~~~
```

```
init.C:933:8: error: invalid user-defined conversion from 'std::atomic<unsigned int>' to 'PUP::able**' [-fpermissive]
     pu(numZerocopyROops);
        ^~~~~~~~~~~~~~~~
```

Reply:
Hmm, interesting. Do these also not work?:
`p(numZerocopyROops);` and `p((UInt)numZerocopyROops);`?

Reply:
```
init.C:933:8: error: invalid conversion from 'std::__atomic_base<unsigned int>::__int_type' {aka 'unsigned int'} to 'CkMigrateMessage*' [-fpermissive]
     pu((UInt)numZerocopyROops);
        ^~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Ok, this is unfortunate.

Reply:
We should be able to add a generic pup routine for std::atomic<T> that loads/stores the value in and out of a bare T, then this code can be simplified.